lib/reek/context/ghost_context.rb

```diff
@@ -0,0 +1,62 @@
+require_relative 'code_context'
+require_relative 'singleton_method_context'
+
+module Reek
+  module Context
+    # Semi-transparent context to represent a metaclass while building the
+    # context tree. This context will not be part of the resulting tree, but
+    # will track context and visibility separately while building is in
+    # progress.
+    class GhostContext
```

Comment:
I thought it needed CodeContext, but then it turned out most of the methods that get used need a different implementation. I'll revisit this decision.